s3/src/main/java/io/aiven/kafka/tiered/storage/s3/S3RemoteStorageManager.java

```diff
@@ -64,7 +64,15 @@ public void configure(final Map<String, ?> configs) {
     private AmazonS3 initS3Client(final AwsClientBuilder.EndpointConfiguration endpointConfiguration) {
         AmazonS3ClientBuilder s3ClientBuilder = AmazonS3ClientBuilder.standard();
         if (endpointConfiguration == null) {
-            s3ClientBuilder = s3ClientBuilder.withRegion(config.s3Region());
+            if (config.s3EndpointUrl().isEmpty()) {
+                s3ClientBuilder = s3ClientBuilder.withRegion(config.s3Region());
+            } else {
+                final AwsClientBuilder.EndpointConfiguration endpointConfig =
+                    new AwsClientBuilder.EndpointConfiguration(
+                        config.s3EndpointUrl().get(),
+                        config.s3Region().getName());
```

Comment:
We should perhaps specify this parameters in tests also and these will allow to simplify all these if statements.

Reply:
Just applied this fixes, let me know how it looks.